test/e2e/scenarios/ciliumeventobserver/validate-ciliumeventobserver-metric.go

```diff
@@ -0,0 +1,89 @@
+package ciliumeventobserver
+
+import (
+	"fmt"
+	"log"
+
+	prom "github.com/microsoft/retina/test/e2e/framework/prometheus"
+)
+
+var (
+	dropCountMetricName = "hubble_drop_total"
+	tcpFlagsMetricName  = "hubble_tcp_flags_total"
+	flowsMetricName     = "hubble_flows_processed_total"
+)
+
+const (
+	destinationKey = "destination"
+	sourceKey      = "source"
+	protcolKey     = "protocol"
+	reasonKey      = "reason"
+	directionKey   = "direction"
+)
+
+type CEODropMetric struct {
+	PortForwardedHubblePort string
+	Source                  string
+	Protocol                string
+	Reason                  string
+	Direction               string
+}
+
+type CEOFlowsAndTCPMetrics struct {
+	PortForwardedHubblePort string
+	Source                  string
+}
+
+func (v *CEODropMetric) Run() error {
+	promAddress := fmt.Sprintf("http://localhost:%s/metrics", v.PortForwardedHubblePort)
+
+	metric := map[string]string{
+		directionKey: v.Direction, reasonKey: PolicyDenied,
```

Comment:
should this be v.Reason instead

Reply:
sicne this is advanced, we want to check the full metric presence, including source or destination

Reply:
hubble dataplane has bug with enrichement so source and destination will be empty unless we change the flow egress context in values.yaml from pod to IP. This will not fix the underlying issue though, so we should probably wait until that is fixed and then update the e2e tests.